compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DocumentationAnalyzer.java

```diff
@@ -241,14 +255,24 @@ public void visit(BLangResource resourceNode) {
 
         validateReturnParameter(resourceNode, null, false);
         validateReferences(resourceNode);
+        validateDeprecationDocumentation(resourceNode.markdownDocumentationAttachment, false, resourceNode.pos, false);
+        validateDeprecatedParametersDocumentation(resourceNode.markdownDocumentationAttachment, resourceNode.pos);
     }
 
     private void validateDeprecationDocumentation(BLangMarkdownDocumentation documentation,
-                                                  boolean isDeprecationAnnotationAvailable, DiagnosticPos pos) {
+                                                  boolean isDeprecationAnnotationAvailable,
+                                                  DiagnosticPos pos, boolean isExpected) {
         if (documentation == null) {
             return;
         }
 
+        BLangMarkDownDeprecationDocumentation deprecationDocumentation =
+                documentation.getDeprecationDocumentation();
+        if (!isExpected && deprecationDocumentation != null) {
+            dlog.error(pos, DiagnosticCode.DEPRECATION_DOCUMENTATION_NOT_ALLOWED);
+            return;
+        }
+
         boolean isDeprecationDocumentationAvailable = false;
         if (documentation.deprecationDocumentation != null &&
```

Comment:
Can't we use the `deprecationDocumentation` variable?

Reply:
Done